diffpy/snmf/optimization.py

```diff
@@ -2,36 +2,39 @@
 import cvxpy as cp
 
 
-def mkr_box(quadratic_coefficient, linear_coefficient, lower_bound, upper_bound):
-    """
+def mkr_box(stretched_component_gram_matrix, linear_coefficient, lower_bound, upper_bound):
+    """ Finds the weightings of stretched component signals
+
     Solves min J(x) = (linear_coefficient)' * x + (1/2) * x' * (quadratic coefficient) * x where lower_bound <= x <=
-    upper_bound and quadratic_coefficient is symmetric positive definite
+    upper_bound and stretched_component_gram_matrix is symmetric positive definite
 
     Parameters
     ----------
-    quadratic_coefficient: 2d array like
-        A symmetric positive definite 2 dimensional array like object.
+    stretched_component_gram_matrix: 2d array like
+      The Gram matrix constructed from the stretched component matrix. It is a square positive definite matrix. It has
+      dimensions C x C where C is the number of component signals.
 
     linear_coefficient: 1d array like
-        A vector
+      The vector containing
 
     lower_bound: 1d array like
-        Represents the lower bounds on the value of x element wise
+      The lower limit on the value of the output element wise. Has length C
 
     upper_bound: 1d array like
-        Represents the upper bounds on the value of x element wise.
+      The upper limit on the values of the output element wise. Has length C.
 
     Returns
     -------
     1d array like
+      The vector containing the weightings of the component signal at a certain moment. Has length C.
 
     Raises
```

Comment:
you are not raising any exception in this function so remove this section.

If there are constraints on allowed values to avoid a crash then mention this in the parameters section.  For example, `The Gram matrix constructed from the stretched component matrix. It has  dimensions C x C where C is the number of component signals and it must be positive definite`

"Must be" implies the code will crash if it isn't"

Reply:
Ok.. I will remove that section and remember this for the future.